### PR TITLE
Initialize pvp/factions library

### PR DESCRIPTION
**Pull request by overlordbleck**
TODO:
- [x] Test

Added functions/methods:
- `pvp.getPvpers()`
- `pvp.getBuilders()`
- `pvp.playerIsInPvp( ply )`
- `ply:isInPvp()`
- `pvp.playerIsInBuild( ply )`
- `ply:isInBuild()`
- `pvp.getPlayerFactionID( ply )`
- `ply:getFactionID()`
- `pvp.getPlayerFactionRank( ply )`
- `ply:getFactionRank()`
- `pvp.isPlayerInFaction( ply )`
- `ply:isInFaction()`

### CODE REVIEW

File: lua/starfall/libs_sh/pvp.lua [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    checktype( self, ply_meta )
+
+    return getPly( self ):GetNWBool( "CFC_PvP_Mode", false )
+end
+
+-- isInBuild
+function pvp_library.playerIsInBuild( ply )
+    checktype( ply, ply_meta )
+
+    return not getPly( ply ):GetNWBool( "CFC_PvP_Mode", false )
```
Comment (samuelWilliams99) [inline lua/starfall/libs_sh/pvp.lua:71]: Is is possible to have this simply call the other method starfall method by not unpacking it?
```suggestion
    return not ply:isInPvp()
```

File: lua/starfall/libs_sh/pvp.lua [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+end
+
+function player_methods:isInBuild()
+    checktype( self, ply_meta )
+
+    return not getPly( self ):GetNWBool( "CFC_PvP_Mode", false )
```
Comment (samuelWilliams99) [inline lua/starfall/libs_sh/pvp.lua:77]: Same as for `playerIsInbuild`

File: lua/starfall/libs_sh/pvp.lua (hunk 1 above)
Comment (samuelWilliams99) [reply lua/starfall/libs_sh/pvp.lua:71]: And if thats the case, could this function and `playerIsInPvp` just call `ply:isInbuild` and `ply:IsInPvP` respectively?

Comment (overlordbleck) [reply lua/starfall/libs_sh/pvp.lua:71]: Unfortunately those methods are only defined on the server :(

Comment (samuelWilliams99) [reply lua/starfall/libs_sh/pvp.lua:71]: ? They're defined in this file, by not calling `getPly` we're running the Starfall functions, not the lua ones

Comment (overlordbleck) [reply lua/starfall/libs_sh/pvp.lua:71]: I'm confused, you're gonna have to elaborate

Comment (samuelWilliams99) [reply lua/starfall/libs_sh/pvp.lua:71]:
```lua
-- isInPvp
function pvp_library.playerIsInPvp( ply )
    checktype( ply, ply_meta )

    return ply:isInPvp()
end

function player_methods:isInPvp()
    checktype( self, ply_meta )

    return getPly( self ):GetNWBool( "CFC_PvP_Mode", false )
end

-- isInBuild
function pvp_library.playerIsInBuild( ply )
    checktype( ply, ply_meta )

    return ply:isInBuild()
end

function player_methods:isInBuild()
    checktype( self, ply_meta )

    return not self:isInPvp()
end
```
So the pvp_library functions just call the starfall player meta methods, and the player meta `isInBuild` just calls the player meta `isInPvp`
Important thing to note here is that I'm not using `getPly`, so the `isInbuild` and `isInPvp` functions I call on them will be the starfall ones (defined in this file), and not the lua ones (defined in factions)

I think you don't even need to `checktype` in `ply:isInbuild`, and the `isInPvp` check type will grab it, but I'd keep it just to be sure/consistent

Comment (overlordbleck) [reply lua/starfall/libs_sh/pvp.lua:71]: I'm not a big fan of this, I can probably just make a function at the top for getting the info

**Pull request merged**